hypervisor/lib/crypto/mbedtls/md.c

```diff
@@ -31,191 +31,158 @@
+static const mbedtls_md_type_t supported_digests[] = {
         MBEDTLS_MD_SHA256,
         MBEDTLS_MD_NONE
 };
 
-const int32_t *mbedtls_md_list( void )
+const mbedtls_md_type_t *mbedtls_md_list( void )
 {
     return( supported_digests );
 }
 
 const mbedtls_md_info_t *mbedtls_md_info_from_type( mbedtls_md_type_t md_type )
 {
-    switch( md_type )
+    const mbedtls_md_info_t *md_info = NULL;
+
+    switch ( md_type )
     {
         case MBEDTLS_MD_SHA256:
-            return( &mbedtls_sha256_info );
+            md_info = &mbedtls_sha256_info;
+            break;
         default:
-            return( NULL );
+            break;
     }
+
+    return md_info;
 }
 
 void mbedtls_md_init( mbedtls_md_context_t *ctx )
 {
-    memset( ctx, 0, sizeof( mbedtls_md_context_t ) );
+    (void) memset( ctx, 0U, sizeof( mbedtls_md_context_t ) );
 }
 
 void mbedtls_md_free( mbedtls_md_context_t *ctx )
 {
-    if( ctx == NULL )
-        return;
-
-    mbedtls_platform_zeroize( ctx, sizeof( mbedtls_md_context_t ) );
-}
-
-int32_t mbedtls_md_clone( mbedtls_md_context_t *dst,
-                      const mbedtls_md_context_t *src )
-{
-    if( dst == NULL || dst->md_info == NULL ||
-        src == NULL || src->md_info == NULL ||
-        dst->md_info != src->md_info )
-    {
-        return( MBEDTLS_ERR_MD_BAD_INPUT_DATA );
+    if ( ctx != NULL ) {
+        (void) mbedtls_platform_zeroize( ctx, sizeof( mbedtls_md_context_t ) );
     }
 
-    dst->md_info->clone_func( dst->md_ctx, src->md_ctx );
-
-    return( 0 );
+    return;
 }
 
 int32_t mbedtls_md_setup( mbedtls_md_context_t *ctx, const mbedtls_md_info_t *md_info )
 {
-    if( md_info == NULL || ctx == NULL )
-        return( MBEDTLS_ERR_MD_BAD_INPUT_DATA );
-
-    ctx->md_info = md_info;
-
-    return( 0 );
-}
-
-int32_t mbedtls_md_starts( mbedtls_md_context_t *ctx )
-{
-    if( ctx == NULL || ctx->md_info == NULL )
-        return( MBEDTLS_ERR_MD_BAD_INPUT_DATA );
-
-    return( ctx->md_info->starts_func( ctx->md_ctx ) );
-}
-
-int32_t mbedtls_md_update( mbedtls_md_context_t *ctx, const uint8_t *input, size_t ilen )
-{
-    if( ctx == NULL || ctx->md_info == NULL )
-        return( MBEDTLS_ERR_MD_BAD_INPUT_DATA );
-
-    return( ctx->md_info->update_func( ctx->md_ctx, input, ilen ) );
-}
-
-int32_t mbedtls_md_finish( mbedtls_md_context_t *ctx, uint8_t *output )
-{
-    if( ctx == NULL || ctx->md_info == NULL )
-        return( MBEDTLS_ERR_MD_BAD_INPUT_DATA );
-
-    return( ctx->md_info->finish_func( ctx->md_ctx, output ) );
-}
+    int32_t ret = 0;
 
-int32_t mbedtls_md( const mbedtls_md_info_t *md_info, const uint8_t *input, size_t ilen,
-            uint8_t *output )
-{
-    if( md_info == NULL )
-        return( MBEDTLS_ERR_MD_BAD_INPUT_DATA );
+    if ( ( md_info == NULL ) || ( ctx == NULL ) ) {
+        ret = MBEDTLS_ERR_MD_BAD_INPUT_DATA;
+    } else {
+        ctx->md_info = md_info;
+    }
 
-    return( md_info->digest_func( input, ilen, output ) );
+    return( ret );
```

Comment:
return ret ?

Reply:
some functions are removed due to not used.  So the patch seems hard to review. The last function is here.
219 uint8_t mbedtls_md_get_size( const mbedtls_md_info_t *md_info )
220 {
221     uint8_t ret = 0U;
222 
223     if ( md_info != NULL ) {
224         ret = (uint8_t) md_info->size;
225     }
226 
227     return ( ret );
228 }